source/util/LockFile.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "LockFile.h"
+#include "../logging/LoggerFactory.h"
+
+#include <csignal>
+#include <stdexcept>
+#include <unistd.h>
+
+using namespace std;
+using namespace Aws::Iot::DeviceClient::Util;
+using namespace Aws::Iot::DeviceClient::Logging;
+
+constexpr char LockFile::TAG[];
+
+LockFile::LockFile(const std::string &filename, const std::string &process) : filename(filename)
+{
+    ifstream fileIn(filename);
+    if (fileIn)
+    {
+        string storedPid;
+        if (fileIn >> storedPid && !(kill(stoi(storedPid), 0) == -1 && errno == ESRCH))
+        {
+            string processPath = "/proc/" + storedPid + "/cmdline";
+            string cmdline;
+            ifstream cmd(processPath.c_str());
+            // check if process contains name
+            if (cmd && cmd >> cmdline && cmdline.find(process) != string::npos)
+            {
+                LOGM_ERROR(TAG, "Pid associated with active process found in lockfile: %s", filename.c_str());
```

Comment:
Include the process name in the log message eg `Pid associated with active process named %s in lockfile: %s"`